src/deriveNativeScriptHash.c

```diff
@@ -0,0 +1,464 @@
+#include "deriveNativeScriptHash.h"
+#include "state.h"
+#include "textUtils.h"
+#include "uiScreens.h"
+
+static ins_derive_native_script_hash_context_t* ctx = &(instructionState.deriveNativeScriptHashContext);
+
+// Helper functions
+
+#define TRACE_WITH_CTX(message, ...) TRACE(message "level = %u, remaining scripts = %u", ##__VA_ARGS__, ctx->level, ctx->complexScripts[ctx->level].remainingScripts)
+#define SAFE_SPRINTF(str, format, ...) {\
+		explicit_bzero(str, SIZEOF(str));\
+		snprintf(str, SIZEOF(str), format, __VA_ARGS__);\
+		ASSERT(strlen(str) + 1 < SIZEOF(str));\
```

Comment:
This is only needed if you don't trust snprintf, but then what is the point of snprintf?

Reply:
Fair point, I removed it :)